src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -0,0 +1,233 @@
+            }
+        }
+
+        public override Symbol ContainingSymbol
+        {
+            get
+            {
+                return _containingTuple;
+            }
+        }
+
+        public override ImmutableArray<CustomModifier> CustomModifiers
+        {
+            get
+            {
+                return _underlyingField.CustomModifiers;
+            }
+        }
+
+        internal override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBeingBound)
+        {
+            return _underlyingField.GetFieldType(fieldsBeingBound);
+        }
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            return _underlyingField.GetAttributes();
+        }
+
+        internal override DiagnosticInfo GetUseSiteDiagnostic()
+        {
+            DiagnosticInfo result = base.GetUseSiteDiagnostic();
+            MergeUseSiteDiagnostics(ref result, _underlyingField.GetUseSiteDiagnostic());
+            return result;
+        }
+
+        public override sealed int GetHashCode()
+        {
+            return Hash.Combine(_containingTuple.GetHashCode(), _tupleFieldId.GetHashCode());
+        }
+
+        public override sealed bool Equals(object obj)
+        {
+            return Equals(obj as TupleFieldSymbol);
+        }
+
+        public bool Equals(TupleFieldSymbol other)
+        {
+            if ((object)other == this)
+            {
+                return true;
+            }
+
+            return (object)other != null && _tupleFieldId == other._tupleFieldId && _containingTuple == other._containingTuple;
+        }
+    }
+
+    /// <summary>
+    /// Represents an element field of a tuple type (such as (int, byte).Item1)
+    /// that is backed by a real field with the same name within the tuple underlying type.
+    /// </summary>
+    internal class TupleElementFieldSymbol : TupleFieldSymbol
+    {
+        private readonly ImmutableArray<Location> _locations;
+
+        public TupleElementFieldSymbol(TupleTypeSymbol container, FieldSymbol underlyingField, int tupleFieldId, Location location)
+            : base(container, underlyingField, tupleFieldId)
+        {
+            _locations = location == null ? ImmutableArray<Location>.Empty : ImmutableArray.Create(location);
+        }
+
+        public override ImmutableArray<Location> Locations
+        {
+            get
+            {
+                return _locations;
+            }
+        }
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
+        {
+            get
+            {
+                return GetDeclaringSyntaxReferenceHelper<CSharpSyntaxNode>(_locations);
+            }
+        }
+
+        public override bool IsImplicitlyDeclared
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        internal override int? TypeLayoutOffset
```

Comment:
Why, if we can implement and test the implementation?

---
In reply to: [61812930](https://github.com/dotnet/roslyn/pull/11006#discussion_r61812930) [](ancestors = 61812930)